src-vue/src/components/PlayButton.vue

```diff
@@ -85,9 +87,64 @@ export default defineComponent({
                 : 'border-radius: 2px';
         }
     },
+    data() {
+        return {
+        percentage: 0,
+        color: '#409EFF',
+        install_or_update: false,
+        status: "unknown",
+        current_downloaded: -1,
+        total_size: -1,
+        };
+    },
     methods: {
-        launchGame() {
+        formatBytes(bytes: number, decimals = 2) {
+            if (bytes === 0) return '0 Bytes';
+            const k = 1000;
+            const dm = decimals < 0 ? 0 : decimals;
+            const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+            const i = Math.floor(Math.log(bytes) / Math.log(k));
+            return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+        },
+        formatText() {
+            if (this.status == "DOWNLOADING") {
+                const current_downloaded_string = this.formatBytes(this.current_downloaded);
+                const total_size_string = this.formatBytes(this.total_size);
+                const status = this.$t("generic.downloading");
+                return `${status}: ${current_downloaded_string}/${total_size_string}`;
+            }
+            if (this.status == "EXTRACTING") {
+                return this.$t("generic.extracting");
+            }
+            return "";
+        },
+        async launchGame() {
+            let unlistenProgress = await appWindow.listen(
+                'northstar-install-download-progress',
+                ({ event, payload }) => {
+                    this.install_or_update = true;
+                    let progress = payload as InstallProgress; // This is bad but don't know how to do it properly
```

Comment:
Why is this "bad"?
If you're talking about the typing, you can type the `listen` generic method invocation with expected answer type:
```typescript
appWindow.listen<InstallType>( ... )
```

Reply:
Technically `InstallProgress` but yes. Thing is, when I try that I get:

> `'InstallProgress' only refers to a type, but is being used as a value here.ts(2693)`

Reply:
Nevermind, I had typo in the syntax. Done properly in 48e7a2d25934f7e2c7873b3a1a291455f13475e0 now ^^